File: backend/app/schemas/folder.py
import uuid
from datetime import datetime

from pydantic import BaseModel, Field


class FolderCreate(BaseModel):
    name: str = Field(min_length=1, max_length=255)
    parent_folder_id: uuid.UUID | None = None
    color: str | None = None
    icon: str | None = None


class FolderUpdate(BaseModel):
    name: str | None = Field(None, min_length=1, max_length=255)
    color: str | None = None
    icon: str | None = None


class FolderMove(BaseModel):
    parent_folder_id: uuid.UUID | None = None


class FolderResponse(BaseModel):
    id: uuid.UUID
    name: str
    owner_id: uuid.UUID
    parent_folder_id: uuid.UUID | None = None
    color: str | None = None
    icon: str | None = None
    is_trashed: bool
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class BreadcrumbItem(BaseModel):
    id: uuid.UUID
    name: str

File: frontend-app/src/types/folder.ts
export interface Folder {
  id: string;
  name: string;
  owner_id: string;
  parent_folder_id: string | null;
  color: string | null;
  icon: string | null;
  is_trashed: boolean;
  created_at: string;
  updated_at: string;
}

export interface FolderContents {
  folders: Folder[];
  documents: import("./document").Document[];
  spreadsheets: import("./spreadsheet").Spreadsheet[];
  presentations: import("./presentation").Presentation[];
}

export interface BreadcrumbItem {
  id: string;
  name: string;
}
